pkg/controller/jobframework/reconciler.go

```diff
@@ -630,7 +630,16 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 	if workloadSliceEnabled(job) {
 		// Start workload-slice schedule-gated pods (if any).
 		log.V(3).Info("Job running with admitted workload slice, start pods.")
-		return ctrl.Result{}, workloadslicing.StartWorkloadSlicePods(ctx, r.client, object)
+		// Check if job implements JobWithPodLabelSelector interface
+		var podLabelSelector string
+		if jobWithPodLabelSelector, ok := job.(JobWithPodLabelSelector); ok {
+			podLabelSelector = jobWithPodLabelSelector.PodLabelSelector()
+			log.V(10).Info("Using PodLabelSelector from job", "podLabelSelector", podLabelSelector)
+		} else {
+			log.V(10).Info("Not using PodLabelSelector from job")
+		}
+
+		return ctrl.Result{}, workloadslicing.StartWorkloadSlicePods(ctx, r.client, object, podLabelSelector)
```

Comment:
Is it possible that the label selector we get here is empty by design? Meaning we would want to StartWorkloadSlicePods with any set of labels? 
If yes then the current logic may clash with this case due to defaulting to using the owner reference UID if the selector is empty.

Reply:
Updated PR, do not need logic using pod selector anymore, can still use owner reference UID.